pkg/controllers/machinesetsync/machineset_sync_controller_test.go

```diff
@@ -156,56 +155,280 @@ var _ = Describe("With a running MachineSetSync controller", func() {
+								HaveField("Reason", Equal("FailedToGetCAPIMachineSet")),
+								HaveField("Message", Equal("error synchronizing CAPI to MAPI, CAPI machineset not found")),
+							))),
+					))
+
+			})
+
 		})
+
+		Context("When the MAPI MachineAuthority is set to Migrating", func() {
+			BeforeEach(func() {
+				By("Creating the CAPI and MAPI MachineSets")
+				mapiMachineSet = mapiMachineSetBuilder.Build()
+				capiMachineSet = capiMachineSetBuilder.Build()
+
+				Expect(k8sClient.Create(ctx, capiMachineSet)).Should(Succeed())
+				Expect(k8sClient.Create(ctx, mapiMachineSet)).Should(Succeed())
+
+				By("Setting the AuthoritativeAPI to Migrating")
+				Eventually(k.UpdateStatus(mapiMachineSet, func() {
+					mapiMachineSet.Status.AuthoritativeAPI = machinev1beta1.MachineAuthorityMigrating
+				})).Should(Succeed())
+			})
+
+			It("should not make any changes", func() {
+				resourceVersion := mapiMachineSet.GetResourceVersion()
+				Consistently(k.Object(mapiMachineSet), timeout).Should(
+					HaveField("ResourceVersion", Equal(resourceVersion)),
+				)
+			})
+		})
+
+		Context("When the MAPI MachineAuthority is not set", func() {
+			BeforeEach(func() {
+				By("Creating the CAPI and MAPI MachineSets")
+				mapiMachineSet = mapiMachineSetBuilder.Build()
+				capiMachineSet = capiMachineSetBuilder.Build()
+
+				Expect(k8sClient.Create(ctx, capiMachineSet)).Should(Succeed())
+				Expect(k8sClient.Create(ctx, mapiMachineSet)).Should(Succeed())
+
+				By("Setting the AuthoritativeAPI to Migrating")
+				Eventually(k.UpdateStatus(mapiMachineSet, func() {
+					mapiMachineSet.Status.AuthoritativeAPI = ""
+				})).Should(Succeed())
+			})
+
+			// I think there may be another case here, where we have had it previously set - synchronized and then it gets unset.
+			// We may want to ensure that we don't change resourceVersion, and that the condition does not change?
+			// Changes to false with a failure reason?
+			//
+			// I'm not sure if we have an API validation that stops this.
+			// TODO: clarify / work out
+
+			It("should not make any changes", func() {
+				resourceVersion := mapiMachineSet.GetResourceVersion()
+				Consistently(k.Object(mapiMachineSet), timeout).Should(
+					HaveField("ResourceVersion", Equal(resourceVersion)),
+				)
+			})
+		})
+
+		Context("when the MAPI MachineSet does not exist and the CAPI MachineSet does", func() {
+			BeforeEach(func() {
+				By("Creating the CAPI MachineSet")
+				capiMachineSet = capiMachineSetBuilder.Build()
+
+				Expect(k8sClient.Create(ctx, capiMachineSet)).Should(Succeed())
+			})
+
+			It("should not make any changes to the CAPI MachineSet", func() {
+				resourceVersion := capiMachineSet.GetResourceVersion()
+				Consistently(k.Object(capiMachineSet), timeout).Should(
+					HaveField("ResourceVersion", Equal(resourceVersion)),
+				)
+			})
+
+			// Do we have another case here where we consistently want to not see a MAPI MachineSet created?
+			// Not sure if this is overkill
+
```

Comment:
I'd say consistently no new MAPI MachineSets seems reasonable yes, in the future we may decide to change that behaviour so we should test it, as we have decided it for now